cdk/lib/mparticle-api.ts

```diff
@@ -26,6 +28,44 @@ export class MParticleApi extends GuStack {
 			},
 		});
 
+		// API Gateway
+		new GuApiGatewayWithLambdaByPath(this, {
+			app: app,
+			targets: [
+				{
+					path: '/data-subject-requests/{requestId}/callback',
+					httpMethod: 'POST',
+					lambda: lambda,
+				},
+			],
+			// Create an alarm
+			monitoringConfiguration: {
+				snsTopicName: `alarms-handler-topic-${this.stage}`,
+				http5xxAlarm: {
+					tolerated5xxPercentage: 1,
+				},
+			},
+		});
+
+		// SQS Queue
+		const mparticleCallbackDeadLetterQueue = new Queue(
+			this,
+			'MparticleApiCallbackDeadLetterQueue',
+			{
+				queueName: `mparticle-api-callback-dead-letter-queue-${this.stage}`,
+				retentionPeriod: Duration.seconds(864000),
+			},
+		);
+
+		new Queue(this, 'MparticleCallbackQueue', {
+			queueName: `mparticle-callback-queue-${this.stage}`,
+			visibilityTimeout: Duration.seconds(3000),
```

Comment:
does this work? or failing that 50*60
```suggestion
			visibilityTimeout: Duration.minutes(50),
```